lib/Sema/CSDiag.cpp

```diff
@@ -2273,6 +2273,27 @@ diagnoseUnviableLookupResults(MemberLookupResult &result, Type baseObjTy,
       } else {
         // Otherwise the static member lookup was invalid because it was
         // called on an instance
+        
+        // Handle enum element lookup on instance type
+        auto lookThroughBaseObjTy = baseObjTy->lookThroughAllAnyOptionalTypes();
+        if (lookThroughBaseObjTy->is<EnumType>()
+            || lookThroughBaseObjTy->is<BoundGenericEnumType>()) {
+          bool triedToAccessEnumElement = false;
+          for (auto cand : result.UnviableCandidates) {
+            ValueDecl *decl = cand.first;
+            if (isa<EnumElementDecl>(decl)) {
+              triedToAccessEnumElement = true;
+              break;
+            }
+          }
+          if (triedToAccessEnumElement) {
+            diagnose(loc, diag::could_not_use_enum_element_on_instance,
```

Comment:
I think, I had an issue with the instantiation of generic parameters, but now that I think about it again, I believe adding a fully qualified type in front should be easily possible. I will add this fix-it together with the changed error message.